consensus/src/proposal/handler.rs

```diff
@@ -42,11 +44,16 @@ impl<D: Database> MsgHandler for ProposalHandler<D> {
     async fn collect(
         &mut self,
         msg: Message,
-        _ru: &RoundUpdate,
+        ru: &RoundUpdate,
         _committee: &Committee,
     ) -> Result<HandleMsgOutput, ConsensusError> {
         // store candidate block
         let p = Self::unwrap_msg(&msg)?;
+
+        if p.candidate.header().timestamp < ru.timestamp() {
```

Comment:
Do you have any intention by putting this check here instead of `ProposalHandler::verify(`?  If so, let's have it as a comment.

Reply:
It should be the same, since both will discard the block. I'm not strongly opinionated on this

Reply:
If `ProposalHandler::verify(` considers it valid (returns true) the message is rebroadcast. (if that is the intention then it's fine)